src/discord/discord.service.ts

```diff
@@ -1,26 +1,71 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { CreateDiscordDto } from './dto/create-discord.dto';
+import { ReadDiscordDto } from './dto/read-discord.dto';
 import { UpdateDiscordDto } from './dto/update-discord.dto';
 
 @Injectable()
 export class DiscordService {
+  private readonly discord: ReadDiscordDto[] = [];
   create(createDiscordDto: CreateDiscordDto) {
-    return 'This action adds a new discord';
+    const discordUser = {
+      id: Date.now(),
+      username: createDiscordDto.username,
+      bio: { ...createDiscordDto.bio },
+      createdOn: new Date(Date.now()),
+      updatedOn: new Date(Date.now()),
+    };
+    this.discord.push(discordUser);
+    return 'User added successfully!';
   }
 
   findAll() {
-    return `This action returns all discord`;
+    return [...this.discord];
   }
 
   findOne(id: number) {
-    return `This action returns a #${id} discord`;
+    const discordUser = this.discord.find((user) => user.id === id);
+    this.DiscordNotFound(discordUser);
+    return { ...discordUser };
   }
 
   update(id: number, updateDiscordDto: UpdateDiscordDto) {
-    return `This action updates a #${id} discord`;
+    const {
+      username,
+      bio: { description, twitter, linkedin, github },
+    } = updateDiscordDto;
+
+    const discordUser = this.discord.find((user) => user.id === id);
+    this.DiscordNotFound(discordUser);
+    const updatedDiscord = { ...discordUser };
+    if (username) {
+      updatedDiscord.username = username;
+    }
+    if (description) {
+      updatedDiscord.bio.description = description;
+    }
+    if (twitter) {
+      updatedDiscord.bio.twitter = twitter;
+    }
+    if (linkedin) {
+      updatedDiscord.bio.linkedin = linkedin;
+    }
+    if (github) {
+      updatedDiscord.bio.github = github;
+    }
+
+    return 'User updated successfully!';
   }
 
   remove(id: number) {
-    return `This action removes a #${id} discord`;
+    const userIndex = this.discord.findIndex((user) => user.id === id);
```

Comment:
`find` would work better here, but still better to use es6 style

It will eventually be replaced by the DB

Reply:
> `find` would work better here.
> 
> It will eventually be replaced by the DB, but still better to use es6 style

I'm using splice in the next line. I am not sure how can we remove an item if we use `find`. How about we use `filter` instead of `findIndex` or `find` and update the whole record.

Reply:
> > `find` would work better here.
> > It will eventually be replaced by the DB, but still better to use es6 style
> 
> I'm using `splice` in the next line. I am not sure how can we remove an item if we use `find`. How about we use `filter` instead of `findIndex` or `find` and update the whole record.

**Note:** `this.discord` is read-only at the moment, we will have to tweak that if we use `filter`

Reply:
sorry I meant `filter` not `find`, but yes `filter` out the one we no longer required would be cleaner